onnx/backend/test/case/node/scaleddotproductattention.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+
+import onnx
+from ..base import Base
+from . import expect
+
+
+def dropout(X, drop_probability=0.0, training_mode=False):  # type: ignore
+    if training_mode is False:
+        return X
+    mask = np.random.uniform(0, 1.0, X.shape) >= drop_probability
+    scale = (1 / (1 - drop_probability))
+    return mask * X * scale
+
+
+class ScaledDotProductAttention(Base):
+    @staticmethod
+    def export() -> None:
+        bsz, tgt_len, src_len, head_dim = 4, 16, 20, 4
+        dropout, training_mode = 0.2, 0
+        query = np.random.randn(bsz, tgt_len, head_dim)
+        key = np.random.randn(bsz, src_len, head_dim)
+        value = np.random.randn(bsz, src_len, head_dim)
+
+        scaling = float(head_dim) ** -0.5
+        attn_output_weights = np.matmul(query, key) / scaling
+
+        t = np.exp(attn_output_weights)
+        attention = t / np.expand_dims(np.sum(t, axis=-1), -1)
+        attention = dropout(attention, dropout, training_mode > 0)
+        output = np.matmul(attention, value).reshape(bsz, tgt_len, head_dim)
+
+        node = onnx.helper.make_node(
+            'MultiHeadAttention',
+            inputs=['query', 'key', 'value'],
+            outputs=['output'],
+            training_mode=training_mode,
+            dropout=dropout
+        )
+
+        expect(node, inputs=[query, key, value],
+               outputs=[output],
```

Comment:
any way to also output attn_output_weights ?

Reply:
There is no confilct between using attn_mask and outputting attention weight.